test/libsolidity/syntaxTests/bound/bound_to_calldata_struct.sol

```diff
@@ -0,0 +1,15 @@
+struct S {
+    uint x;
+}
+
+library L {
+    function f(S calldata) internal pure {}
+}
+
+contract C {
+    using L for S;
+
+    function run(S calldata _s) external pure {
+        _s.f();
+    }
+}
```

Comment:
The error is in the codegen but if happens in semantic tests too. I wrote only semantic ones initially and added this one later but maybe syntax ones would be enough.